### ISSUE TITLE
Multiple identical sorts not elided by optimizer

### ISSUE THREAD

**Issue by AlecZorab** [pola-rs/polars#15447]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
(
    pl.DataFrame({"symbol": ["a", "a", "b", "b"]})
    .lazy()
    .sort("symbol")
    .sort("symbol")
    .explain(optimized=True)
)
```

### Log output

```shell
SORT BY [col("symbol")]
  SORT BY [col("symbol")]
    DF ["symbol"]; PROJECT */1 COLUMNS; SELECTION: "None"
```

### Issue description

 when performing the same sort twice in, a row one of them could be dropped

### Expected behavior

expect output to look like 
```
  SORT BY [col("symbol")]
    DF ["symbol"]; PROJECT */1 COLUMNS; SELECTION: "None"
```

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.18
Index type:           UInt32
Platform:             macOS-14.2.1-arm64-arm-64bit
Python:               3.12.2 (main, Feb  6 2024, 20:19:44) [Clang 15.0.0 (clang-1500.1.0.2.5)]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          3.0.0
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            <not installed>
fsspec:               <not installed>
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           <not installed>
nest_asyncio:         1.6.0
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               2.2.1
pyarrow:              15.0.2
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

**Comment by AlecZorab** [comment]
stretch goal:
```python
(
    pl.DataFrame({"symbol": ["a", "a", "b", "b"], "Timestamp": [1, 2, 1, 2], "val": [1, 2, 3, 4]})
    .lazy()
    .sort("symbol")
    .select(["symbol", "val"])
    .sort("symbol")
    .explain(optimized=True)
)
```

```
SORT BY [col("symbol")]
   SELECT [col("symbol"), col("val")] FROM
    SORT BY [col("symbol")]
      DF ["symbol", "Timestamp", "val"]; PROJECT 2/3 COLUMNS; SELECTION: "None"'
```

I think one of these sorts can be skipped too

**Comment by lukemanley** [comment]
I believe this was fixed by #15573 and can now be closed. Both of the examples above are now optimized to sort only once:

example 1: 
```
SORT BY [col("symbol")]
  DF ["symbol"]; PROJECT */1 COLUMNS; SELECTION: None
```

example 2:
```
SORT BY [col("symbol")]
  DF ["symbol", "Timestamp", "val"]; PROJECT 2/3 COLUMNS; SELECTION: None
```
